pytorch_lightning/utilities/distributed.py

```diff
@@ -399,3 +399,37 @@ def _collect_states_on_rank_zero(state: Dict[str, Any]) -> Dict[int, Any]:
     if not distributed_available():
         return {0: state}
     return {rank: _broadcast_object_list(state, rank) for rank in range(torch.distributed.get_world_size())}
+
+
+class _BatchNormXd(torch.nn.modules.batchnorm._BatchNorm):
+    def _check_input_dim(self, input):
+        # The only difference between BatchNorm1d, BatchNorm2d, BatchNorm3d, etc
+        # is this method that is overwritten by the subclass.
+        # Here, we are bypassing some tensor sanity checks and trusting that the user
+        # provides the right input dimensions at inference.
+        return
```

Comment:
An alternative to this could be to overwrite the SyncBatchnorm to also store the dimensionality and being able to correctly recover lateron since otherwise typechecks on the userside might not work as expected when they check against a specific BN version (1d, 2d, 3d) .

Reply:
yes, but you don't need to store the class but only the int (1,2,3)...

Reply:
That would require us to copy and modify the implementation of SyncBatchNorm.convert_sync_batch_norm to store this information. I would prefer to wait and see what pytorch replies on my issue here https://github.com/pytorch/pytorch/issues/69972 before investigating that. What do you think?

Reply:
Yes, that's true. Fine with me :)